src/main/java/com/google/sps/servlets/YoutubeServlet.java

```diff
@@ -14,12 +14,68 @@
 
 package com.google.sps.servlets;
 
+import com.google.api.client.googleapis.javanet.GoogleNetHttpTransport;
+import com.google.api.client.googleapis.json.GoogleJsonResponseException;
+import com.google.api.client.http.javanet.NetHttpTransport;
+import com.google.api.client.json.JsonFactory;
+import com.google.api.client.json.jackson2.JacksonFactory;
+import com.google.api.services.youtube.model.CommentThreadListResponse;
+import com.google.api.services.youtube.YouTube;
+import java.io.IOException;
+import java.security.GeneralSecurityException;
+import java.util.Arrays;
+import java.util.Collection;
 import javax.servlet.annotation.WebServlet;
 import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
 
 /**
- * Servlet that fetch from Youtube Server.
+ * Servlet that fetches from Youtube Server.
  */
-@WebServlet("/userLogIn")
+@WebServlet("/YouTubeComments")
 public class YoutubeServlet extends HttpServlet {
+  // TODO: obtain actual API Key
+  private static final String DEVELOPER_KEY = "OUR_API_KEY"; 
+  private static final String APPLICATION_NAME = "SAY";
+  private static final JsonFactory JSON_FACTORY = JacksonFactory.getDefaultInstance();
+
+  /**
+   * Applies parameters to the YouTube API, then extracts comments.
+   * URL is the only true variable; for this application we will always want
+   * order to be relevance and max results to be 100, the API's arbitrary limit
+   */
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    try {
+      YouTube youtubeService = getService();
+      String url = request.getParameter("url");
+      // Define and execute the API request
+      YouTube.CommentThreads.List commentRequest = youtubeService.commentThreads()
+          .list("snippet,replies");
+      CommentThreadListResponse commentResponse = commentRequest.setKey(DEVELOPER_KEY)
+          .setVideoId(url)
+          .setOrder("relevance")
+          .setMaxResults(100L)
+          .execute();
+      response.getWriter().println(commentResponse);
+    } catch (GeneralSecurityException | GoogleJsonResponseException e) { 
+        System.err.println("Error in retrieving YouTube Comments!");
```

Comment:
Also nit: referring to the try/catch statement example here, the indentation should be at 'c' : https://google.github.io/styleguide/javaguide.html#s4.1.2-blocks-k-r-style